crates/astria-gossipnet/src/network_stream.rs

```diff
@@ -0,0 +1,271 @@
+use std::{
+    pin::Pin,
+    str::FromStr,
+    task::{
+        Context,
+        Poll,
+    },
+};
+
+use futures::StreamExt;
+#[cfg(feature = "dht")]
+use libp2p::kad::{
+    record::Key,
+    {
+        KademliaEvent,
+        QueryResult,
+    },
+};
+#[cfg(feature = "mdns")]
+use libp2p::mdns;
+use libp2p::{
+    gossipsub::{
+        self,
+        Message,
+        TopicHash,
+    },
+    kad::{
+        AddProviderError,
+        Addresses,
+        GetClosestPeersError,
+        GetProvidersError,
+    },
+    swarm::SwarmEvent,
+    Multiaddr,
+    PeerId,
+};
+use tracing::{
+    debug,
+    warn,
+};
+
+use crate::network::{
+    GossipnetBehaviourEvent,
+    Network,
+};
+
+#[derive(Debug)]
+pub enum Event {
+    NewListenAddr(Multiaddr),
+    Message(Message),
+    #[cfg(feature = "mdns")]
+    MdnsPeersConnected(Vec<PeerId>),
+    #[cfg(feature = "mdns")]
+    MdnsPeersDisconnected(Vec<PeerId>),
+    PeerConnected(PeerId),
+    PeerSubscribed(PeerId, TopicHash),
+    #[cfg(feature = "dht")]
+    FoundProviders(Option<Key>, Option<Vec<PeerId>>),
+    #[cfg(feature = "dht")]
+    GetProvidersError(GetProvidersError),
+    #[cfg(feature = "dht")]
+    Providing(Key),
+    #[cfg(feature = "dht")]
+    ProvideError(AddProviderError),
+    #[cfg(feature = "dht")]
+    RoutingUpdated(PeerId, Addresses),
+    #[cfg(feature = "dht")]
+    FoundClosestPeers(Vec<PeerId>),
+    #[cfg(feature = "dht")]
+    GetClosestPeersError(GetClosestPeersError),
+}
+
+impl futures::Stream for Network {
```

Comment:
This stream impl is in an odd state between neither being completely high level nor low level. It's partially exposing runtime details, while silently dropping others.

As a consumer it seems to me I should be mainly (only?) interested in `Message`, and I am not sure what I should do with the other variants. Silently drop them?

I believe we could simplify this interface to just yield messages and for the other events just create log entries.

Reply:
most of these I used in the tests, things like `NewListenAddr` are useful for consumers, but I'll remove the ones that aren't really used

Reply:
No need to refactor in this PR, but IMO creating types that are only meant to be used is a code smell.